### PR TITLE
support data plane recording

### PR DESCRIPTION
**Pull request by jordan13game**

### CODE REVIEW

**Review by leniatgh — APPROVED** [review]
Looks great! Please add some test cases

**Comment by leniatgh** [comment]
Some suggestions:
 - [ ] Can we output the schema and source recording comments? like:
```
# yaml-language-server: $schema=https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/documentation/api-scenario/references/v1.2/schema.json

# Generated from test-proxy recording in:
# https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/cognitivelanguage/Azure.AI.Language.QuestionAnswering/tests/SessionRecords/QuestionAnsweringAuthoringClientLiveTests
```
 - [ ] We could skip generating step name (`step`) (loader supports assigning name according to operationId), as the generated scenario will be simpler

File: lib/apiScenario/apiScenarioSchema.ts [hunk 1]
```diff
@@ -407,7 +407,7 @@ export const ApiScenarioDefinition: Schema & {
           minProperties: 1,
           additionalProperties: false,
           patternProperties: {
-            "^([0-9]{3})$": {
+            "^([0-9][0-9x]{2})$": {
```
Comment (leniatgh) [inline lib/apiScenario/apiScenarioSchema.ts]:
```suggestion
            "^([2-5][0-9]{2})|([245]xx)$": {
```

File: lib/apiScenario/gen/testRecordingApiScenarioGenerator.ts [hunk 2]
```diff
@@ -134,12 +137,22 @@ export class TestRecordingApiScenarioGenerator {
...
+    for (const p of recordingPaths) {
+      path += `# ${p}\n`;
+    }
+
     for (const { testDef, filePath } of testDefToWrite) {
-      await this.testResourceLoader.writeTestDefinitionFile(filePath, testDef);
+      const fileContent =
+        "# yaml-language-server: $schema=https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/documentation/api-scenario/references/v1.2/schema.json\n\n" +
+        "# Generated from test-proxy recording in:\n" +
+        `${path}` +
```
Comment (leniatgh) [inline lib/apiScenario/gen/testRecordingApiScenarioGenerator.ts:152]:
```suggestion
        `# ${path}` +
```

File: test/apiScenario/fixtures/specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/1.0/scenarios/test.yaml [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+          value: test_value
+          content_type: test_content_type
+          tags:
+            tag1: value1
+            tag2: value2
+      If-None-Match:
+        type: string
+        value: '*'
+    steps:
+      - step: PutKeyValue
```
Comment (leniatgh) [inline test/apiScenario/fixtures/specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/1.0/scenarios/test.yaml]: what's this file for? `step` is still there

File: lib/apiScenario/gen/testRecordingApiScenarioGenerator.ts (hunk 2 above)
Comment (jordan13game) [reply lib/apiScenario/gen/testRecordingApiScenarioGenerator.ts:152]: you can see the code above

File: test/apiScenario/fixtures/specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/1.0/scenarios/test.yaml (hunk 3 above)
Comment (jordan13game) [reply test/apiScenario/fixtures/specification/appconfiguration/data-plane/Microsoft.AppConfiguration/stable/1.0/scenarios/test.yaml]: for encode test, I forgot to remove `step`

File: lib/apiScenario/gen/testRecordingApiScenarioGenerator.ts (hunk 2 above)
Comment (leniatgh) [reply lib/apiScenario/gen/testRecordingApiScenarioGenerator.ts:152]: i see

File: lib/apiScenario/apiScenarioSchema.ts [hunk 4]
```diff
@@ -407,7 +407,7 @@ export const ApiScenarioDefinition: Schema & {
           minProperties: 1,
           additionalProperties: false,
           patternProperties: {
-            "^([0-9]{3})$": {
+            "^([2-5][0-9]{2})|([245]xx)$": {
```
Comment (leniatgh) [inline lib/apiScenario/apiScenarioSchema.ts:410]: please submit PR to update schema.json in spec repo

**Review by leniatgh — APPROVED** [review]

**Pull request merged**